donkeycar/parts/datastore_v2.py

```diff
@@ -0,0 +1,367 @@
+import atexit
+import json
+import mmap
+import os
+import time
+from collections import namedtuple
+from pathlib import Path
+
+
+class Seekable(object):
+    '''
+    A seekable file reader, writer which deals with newline delimited records. \n
+    This reader maintains an index of line lengths, so seeking a line is a O(1) operation.
+    '''
+
+    def __init__(self, file, method='a+', line_lengths=list()):
+        self.method = method
+        self.line_lengths = list()
+        self.file = open(file, self.method)
+        if len(line_lengths) <= 0:
```

Comment:
`== 0` ?

Reply:
I like my conditionals to be conclusive :smile:. Just being defensive.